mempool/cat/reactor.go

```diff
@@ -263,6 +260,7 @@ func (memR *Reactor) ReceiveEnvelope(e p2p.Envelope) {
 				// We broadcast only transactions that we deem valid and actually have in our mempool.
 				memR.broadcastSeenTx(key)
 			}
+			schema.WriteMempoolTx(memR.traceClient, e.Src.ID(), key[:], schema.Download)
```

Comment:
This could potentially lead to problems with the visualising tool because it may look like a transaction was sent but never reached it's target but in reality it did reach the target node but was rejected at a later point. Also this doesn't capture the timeliness of the message. For example I want to know the difference between when a transaction was received and when the node sends the `HasTx` message out to its peers